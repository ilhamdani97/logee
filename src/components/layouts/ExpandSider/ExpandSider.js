import React, { useEffect, useState } from 'react';
import PropTypes from 'prop-types';
import { useLocation, Link } from 'react-router-dom';
import styles from './styles.scoped.css';

const ExpandSider = ({ data, index }) => {
  const [expand, setExpand] = useState({ active: -1, open: -1 });
  const clickExpand = (idx) => () => {
    setExpand({ ...expand, open: expand.open === idx ? -1 : idx });
  };
  const { length } = data.children;
  const { pathname,search } = useLocation();

  useEffect(() => {

    const indexSearch = search.split('&')[0];
    const activeChild = data.children.findIndex(i => i.to === pathname+indexSearch);

    setExpand({ open: expand.open, active: activeChild });
  }, [pathname, search]);

  return (
    <div className={styles['nav-item']} id={`sider-parent-${index}`}>
      <section onClick={clickExpand(index)}>
        <data.icon />
        {data.name}
      </section>
      <ul style={{ maxHeight: `${3 * length}rem` }}>
        {data.children.map((c, cIdx) => {
          const isActive = expand.active === cIdx;
          return (
            <li className={c.disabled ? styles.disable : ''} key={cIdx}>
              <Link className={(isActive && !c.disabled) ? styles.active : ''} to={c.to}>{c.name}</Link>
            </li>
          );
        })}
      </ul>
    </div>
  );
};

ExpandSider.defaultProps = {
  data: {},
  index: null,
};

ExpandSider.propTypes = {
  data: PropTypes.object,
  index: PropTypes.number,
};

export default ExpandSider;


