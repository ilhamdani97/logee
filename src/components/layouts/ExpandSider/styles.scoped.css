.nav-item {
  color: var(--text-color);
  font-size: 0.875rem;
  font-weight: 600;
  line-height: 1.5rem;
  margin-bottom: 0.5rem;
}

.nav-item > section {
  align-items: center;
  cursor: pointer;
  display: flex;
  padding: 0.5rem;
  position: relative;
  transition: background-color 0.25s;
  color: var(--black-4);
}

.nav-item > section > svg {
  margin-right: 1rem;
  width: 1.5rem;
  height: 1.5rem;
}

.nav-item > section > img {
  margin-left: auto;
  transition: transform 0.25s;
  width: 1.5rem;
  height: 1.5rem;
}

.nav-item > section > img.open {
  transform: rotate(-180deg);
}

.nav-item > ul {
  overflow-y: hidden;
  position: relative;
  transition: max-height 0.25s;
}

.nav-item > ul > li {
  margin-bottom: 0.5rem;
}

.nav-item > ul > li > a {
  display: block;
  padding: 0.5rem 3rem;
  transition: background-color 0.25s;
}

.nav-item > ul > li > a.active {
  background-color: var(--grey-color-3);
  border-radius: 0.1875rem;
  color: var(--primary-color);
}
