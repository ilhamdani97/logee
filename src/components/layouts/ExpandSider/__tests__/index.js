import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import ExpandSider from '../ExpandSider';


describe('src/pages/ExpandSider', () => {
  const data = {
    'name': 'Laporan',
    'icon': {},
    'children': [
      {
        'name': 'Pemesanan',
        'to': '/pemesanan'
      },
      {
        'name': 'Pembayaran',
        'to': '/pembayaran'
      },
      {
        'name': 'Penagihan',
        'to': '/penagihan'
      },
      {
        'name': 'Internal B2B',
        'to': '/internalB2B'
      }
    ]
  };
  test('render', () => {
    const shallow = new ShallowRenderer();
    const tree = shallow.render(<ExpandSider data={data} index={1}/>);
    const expanded = tree.props.children[0].props;
    expanded.onClick();
  });

  test('render close', () => {
    const shallow = new ShallowRenderer();
    const tree = shallow.render(<ExpandSider data={data} index={-1}/>);
    expect(tree).toMatchSnapshot();
  });
});
