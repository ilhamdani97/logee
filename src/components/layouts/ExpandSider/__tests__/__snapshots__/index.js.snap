// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/pages/ExpandSider render close 1`] = `
<div
  id="sider-parent--1"
>
  <section
    onClick={[Function]}
  >
    <UNDEFINED />
    Laporan
  </section>
  <ul
    style={
      Object {
        "maxHeight": "12rem",
      }
    }
  >
    <li
      className=""
    >
      <Link
        className=""
        to="/pemesanan"
      >
        Pemesanan
      </Link>
    </li>
    <li
      className=""
    >
      <Link
        className=""
        to="/pembayaran"
      >
        Pembayaran
      </Link>
    </li>
    <li
      className=""
    >
      <Link
        className=""
        to="/penagihan"
      >
        Penagihan
      </Link>
    </li>
    <li
      className=""
    >
      <Link
        className=""
        to="/internalB2B"
      >
        Internal B2B
      </Link>
    </li>
  </ul>
</div>
`;
